### PR TITLE
Added SPM support

### PR DESCRIPTION
**Pull request by 3a4oT**
Added SPM support starting from Swift 5.3 (since it supports resources). Some notes:

- SPM tests not passing when you run it from a command line (SPM(bug)? gets the wrong bundle path) but passing if you use Xcode in SPM edit mode(just open as a package instead of .xcodeproj).

- dependency strategy for `PINOperation` set as - branch `master`. I believe it should be set - from `PINOperation 1.2.0` but unfortunately, it was tagged as `1.2` and SPM can not resolve it. Can the maintainers re-tag/new it with proper sem-ver?

### CODE REVIEW

File: Tests/PINDiskCache+PINCacheTests.m [hunk 1]
```diff
@@ -15,7 +15,7 @@ @implementation PINDiskCache (PINCacheTests)
 - (void)setTtlCacheSync:(BOOL)ttlCache
 {
     [self lock];
-        self->_ttlCache = ttlCache;
+    [self setValue:@(ttlCache) forKey:@"_ttlCache"];
```
Comment (3a4oT) [inline Tests/PINDiskCache+PINCacheTests.m:18]: Fixed build issue when testing package in Xcode.

**Comment by xezero** [comment]
This would be fantastic if this got merged! :)

**Comment by 3a4oT** [comment]
@garrettmoon ping...

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -37,15 +37,25 @@ jobs:
...
+  SPM-Build:
+    runs-on: macos-latest
+    env:
+      DEVELOPER_DIR: /Applications/Xcode_12_beta.app/Contents/Developer
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Verify that PINCache can be build by SPM
+        run: xcrun swift build
```
Comment (garrettmoon) [inline .github/workflows/ci.yaml:61]: Do you mind moving this to the Makefile and calling `make spm` here?

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -37,15 +37,25 @@ jobs:
...
-      - uses: actions/checkout@v1
+      - uses: actions/checkout@v2
       - name: Update
         run: carthage update --no-use-binaries --no-build
       - name: Build
         run: carthage build --no-skip-current
+  SPM-Build:
+    runs-on: macos-latest
+    env:
+      DEVELOPER_DIR: /Applications/Xcode_12_beta.app/Contents/Developer
```
Comment (garrettmoon) [inline .github/workflows/ci.yaml:55]: Can this be updated now that 12 is released?

**Review by garrettmoon — COMMENTED** [review]
Do you mind adding a command to the make file as well

**Comment by 3a4oT** [comment]
@garrettmoon, I've addressed your feedback. Thanks!

**Comment by 3a4oT** [comment]
@garrettmoon  This seems to be a sporadic failure, I can confirm that from time to time the same test fails locally. Can we trigger tests again?

**Review by garrettmoon — APPROVED** [review]
Looks good to me! Thanks so much for adding the support for SPM!

**Pull request merged**